### PR TITLE
box: fix `fselect()` behavior on binary data

### LINKED ISSUE #7040: fselect fails while parsing UUID in binary format
Tarantool 2.10.0-beta2-229-g1b1ddd226

Steps to reproduce:

```lua
tarantool> box.schema.create_space('test'):create_index('pk')
tarantool> box.space.test:format({{name='id', type='unsigned'}, {name='uuid'}})
tarantool> box.space.test:replace{1, require('uuid').bin()}
---
- [1, !!binary rmw8FcPVLUmu/NUcJBoBmg==]
...

tarantool> box.space.test:select()
---
- - [1, !!binary rmw8FcPVLUmu/NUcJBoBmg==]
...

tarantool> box.space.test:fselect()
---
- error: 'builtin/box/schema.lua:2060: bad argument #2 to ''max'' (number expected,
    got nil)'
...
```

### PR DESCRIPTION
**Pull request by Gumix**
Currently it throws an error when encounter binary data,
use Base64 encoding instead.

Closes #7040

### CODE REVIEW

File: test/engine/select.result [hunk 1]
```diff
@@ -3721,3 +3721,30 @@ s:fselect(nil, nil, {columns={4, 3, 'name', 'id'}})
...
+---
+...
+s:fselect({}, nil, {max_width=100})
+---
+- - ​+-----+-----+---------------------------+---------------------------+------------------------+-----+
+  - ​| id  |name |           more            |           col4            |          col5          |col6 |
+  - ​+-----+-----+---------------------------+---------------------------+------------------------+-----+
+  - ​|  1  |  2  |            "😀"            |             4             |Im5vbi1wcmludGFibGU6IIAi|  5  |
+  - ​|  3  |  4  |           true            |WzEsWzMseyJhYSI6MSwiYmIiOiJ|                        |     |
+  - ​|  7  |  8  |IkxvbmcgYmluYXJ5IGRhdGEgXHU|           "sdf"           |   "dsgfsdgsegasges"    |     |
```
Comment (locker) [inline test/engine/select.result:3748]: IMO dumping binary data encoded in base64 without any specification looks confusing. Maybe, add `!!binary` tag, as we do in Console/YaML? Come to think of it, is there any point altogether in printing binary data to the console? Maybe, just print something like `<binary>` instead?

File: src/box/lua/schema.lua [hunk 2]
```diff
@@ -2075,6 +2075,20 @@ base_index_mt.fselect = function(index, key, opts, fselect_opts)
         num_cols = math.max(num_cols, #tab[i])
     end
 
+    -- The JSON encoder above passes through binary data unmodified.
+    -- Use Base64 encoding for such strings.
+    local digest = require('digest')
+    for j = 1,num_cols do
+        for i = 1,num_rows do
+            if tab[i][j] then
+                local _, err = utf8.len(tab[i][j])
```
Comment (locker) [inline src/box/lua/schema.lua:2084]: What happens if binary data comprises a valid, but non-printable UTF8 string? Maybe, we should mask such strings, too, as we do in Console/YaML:

https://github.com/tarantool/tarantool/blob/5557c47a39f36c1d00291050854fa9cbf1382ae1/third_party/lua-yaml/lyaml.cc#L667

**Review by locker — COMMENTED** [review]
Please don't set the `full-ci` label until you get all approvals, unless you deliberately want to run tests on all platforms (e.g. patch platform-dependent code), so as not to overload CI.

File: test/engine/select.result [hunk 3]
```diff
@@ -3721,3 +3721,30 @@ s:fselect(nil, nil, {columns={4, 3, 'name', 'id'}})
...
+...
+_ = s:update(7, {{'=', 'more', string.fromhex(long_bin)}})
+---
+...
+s:fselect({}, nil, {max_width=100})
+---
+- - ​+-----+-----+---------------------------+---------------------------+------------------------+-----+
+  - ​| id  |name |           more            |           col4            |          col5          |col6 |
+  - ​+-----+-----+---------------------------+---------------------------+------------------------+-----+
+  - ​|  1  |  2  |            "😀"            |             4             |Im5vbi1wcmludGFibGU6IIAi|  5  |
```
[the remark this answers is not in the document]
Comment (Gumix) [reply test/engine/select.result:3746]: This issue is known: https://github.com/tarantool/tarantool/blob/master/test/engine/gh-6165-multibyte-fselect.result#L95

File: src/box/lua/schema.lua (hunk 2 above)
Comment (Gumix) [reply src/box/lua/schema.lua:2084]: Valid non-printable chars are escaped by `json.encode` above.

File: test/engine/select.result (hunk 1 above)
Comment (Gumix) [reply test/engine/select.result]: Done

**Review by locker — APPROVED** [review]

**Review by alyapunov — APPROVED** [review]

**Pull request merged**